src/main/java/com/hubspot/ringleader/watcher/PersistentWatcher.java

```diff
@@ -141,13 +148,9 @@ private synchronized void fetch(final boolean backgroundFetch) {
 
       int version = stat.getVersion();
       int previousVersion = lastVersion.getAndSet(version);
+
       if (version != previousVersion) {
         notifyListeners(Event.nodeUpdated(stat, data));
-
-        if (previousVersion != -1 && backgroundFetch) {
```

Comment:
I struggled to understand this conditional 😕 

On the initial fetch previousVersion is always -1, but also `backgroundFetch` was always true. On successive calls previousVersion should not be -1, and backgroundFetch would be false. In any case I don't understand why we should use this to determine if the watcher stopped.

Reply:
Thanks! I hadn't seen that backgroundFetch was also true in the every-ten-minutes invocation of fetch. I just added f499c54908de703dd81475fea0306a9326506f5c to prevent the periodic invocation being scheduled every time `fetchInBackground` is called, and also to hopefully make this behaviour clearer.

I'd like to think we can rely on curator and this check isn't necessary, so I added an error-level log statement so we can determine if this type of replacement happens in practice.

Note there's a race condition that I don't think we can avoid here - our scheduled check might run just before curator fires its changed event. It seems unlikely to happen though because ten minutes is pretty large compared to the few milliseconds it hopefully takes for an event to be delivered.

Reply:
The ZooKeeper docs seem to suggest that you won't see the write until the watcher fires:
https://zookeeper.apache.org/doc/r3.1.2/zookeeperProgrammers.html#sc_WatchGuarantees
> A client will see a watch event for a znode it is watching before seeing the new data that corresponds to that znode.

But because they're using the same single-threaded executor I guess it's possible for the watcher event to be queued on the executor behind this task and replace the curator unnecessarily. At least this doesn't cause a problem for correctness as far as I can tell